### PR TITLE
Fix MacOS assemble workflows

### PR DESCRIPTION
**Pull request by reta**
### Description
Fix MacOS assemble workflows

### Related Issues
See please https://opensearch.slack.com/archives/C0561HRK961/p1725526478827459

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — APPROVED** [review]
LGTM lets see if it passes.

**Review by jed326 — APPROVED** [review]
Thanks @reta !

**Review by dblock — APPROVED** [review]

File: .github/workflows/assemble.yml [hunk 1]
```diff
@@ -30,11 +30,8 @@ jobs:
       - name: Setup docker (missing on MacOS)
         id: setup_docker
         if: runner.os == 'macos'
-        uses: douglascamata/setup-docker-macos-action@main
-        continue-on-error: true
-        with:
-          upgrade-qemu: true
-          colima:  v0.6.8
+        run: |
+          exit 0;
```
Comment (reta) [inline .github/workflows/assemble.yml:34]: @dblock @jed326 the quick fixes I've tried did not succeed :(, making MacOS `assemble` a noop for now and will work through the day to fix it

Comment (harshavamsi) [reply .github/workflows/assemble.yml:34]: thanks @reta for helping here! debugging github actions is hard

**Review by jed326 — APPROVED** [review]

File: .github/workflows/assemble.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/assemble.yml:34]: Sounds good, open an issue please so we don't lose it?

**Review by dblock — APPROVED** [review]

File: .github/workflows/assemble.yml (hunk 1 above)
Comment (reta) [reply .github/workflows/assemble.yml:34]: :100:

Comment (reta) [reply .github/workflows/assemble.yml:34]: https://github.com/opensearch-project/OpenSearch/issues/15743

**Comment by cwperks** [comment]
FYI its possible to set some CI checks to make them not required through settings by updating Branch protection rules

**Pull request merged**